docs/advanced-topics/checksum-support.rst

```diff
@@ -37,6 +50,20 @@ Descriptions for the two endpoints of note are as follows:
 
 Just like the file endpoints, the id parameter used in the endpoints above can be found on an entry's public page; underneath the Info tab, look for the bolded words **TRS**.
 
+To verify a checksum as reported by the Dockstore API matches what you download from the Docker registry first find the checksum
+and image path using one of the above methods for the image you would like to verify. Then download the image using the
+Docker CLI client.
+
+::
+
+    docker pull quay.io/briandoconnor/dockstore-tool-md5sum:1.0.4
+
+When the download has completed a Digest is provided in the terminal output. This should match the checksum provided
+by the Dockstore API.
+
+Depending on your execution engine, the image may be downloaded using another technique and may not have its checksum
```

Comment:
This sounds ominous and I don't quite get it. Are you saying the image can have a different digest based on how it's downloaded?

Reply:
It could be made more clear. I'll take another try. I'm trying to say "this can perform some verification but its not the same as the execution engine verifying the digest". Since we don't have control over them it's technically possible cromwell could get a MITM attack and this verification wouldn't have proved anything. They don't check to see if the digest on dockstore is the same as what is on Dockerhub/quay and so especially in the case of a branch its possible they could diverge.